File: iOS/IssueTracker/IssueTracker/Model/Label.swift
//
//  Label.swift
//  IssueTracker
//
//  Created by 박성민 on 2020/10/28.
//

import Foundation

struct Label {
    let labelId: String
    let name: String
    let description: String
    let color: String
    
    enum CodingKeys: String, CodingKey {
        case labelId = "label_id"
        case name
        case description
        case color
    }
}

File: web-FE/src/api/label.js
import instance from './instance';

export const getAllLabels = async () => {
  const url = '/api/label';
  try {
    const result = await instance.get(url);
    const { labelArray } = result.data;
    return labelArray;
  } catch (error) {
    console.error(error);
  }
};

export const createLabel = async (name, description, color) => {
  const url = '/api/label';
  const body = { name, description, color };
  try {
    await instance.post(url, body);
  } catch (error) {
    console.error(error);
  }
};

export const updateLabel = async (id, name, description, color) => {
  const url = `/api/label/${id}`;
  const body = { label_name: name, description, color };
  try {
    await instance.put(url, body);
  } catch (error) {
    console.error(error);
  }
};

export const deleteLabel = async (labelId) => {
  const url = `/api/label/${labelId}`;
  try {
    await instance.delete(url);
  } catch (error) {
    console.error(error);
  }
};
